src/WebServer.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2021, Kung-Yao Hsieh, Shang-Chen LIN, Yi-Hang Xie. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef HTTPLIB_SRC_WEBSERVER_H
+#define HTTPLIB_SRC_WEBSERVER_H
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#define DEFAULT_IP "127.0.0.1"
+#define DEFAULT_PORT 80
+// 
+class HttpListener;
+class SocketSession;
+
+/** @brief Manage all web-related jobs. */
+class WebServer {
+private:
+    /** @brief Manage all web-related jobs. */
+    static std::shared_ptr<WebServer> web_;
```

Comment:
Please add Get() for singleton.
Consider using raw pointer for singleton.
Please add comment on this being a singleton.

Reply:
we will add a manager class between document and socket later, so we remove class instance here.